public doc on github support process (#1981)
Azure/azure-sdk: docs/policies/github-support-process.md

The process as described here currently applies to the following repositories:

- [Azure/azure-sdk-for-net](https://github.com/Azure/azure-sdk-for-net)
- [Azure/azure-sdk-for-java](https://github.com/Azure/azure-sdk-for-java)
- [Azure/azure-sdk-for-python](https://github.com/Azure/azure-sdk-for-python)
- [Azure/azure-sdk-for-js](https://github.com/Azure/azure-sdk-for-js)
- [Azure/azure-sdk-for-ruby](https://github.com/Azure/azure-sdk-for-ruby)
- [Azure/azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go)
- [Azure/azure-powershell](https://github.com/Azure/azure-powershell)
- [Azure/azure-cli](https://github.com/Azure/azure-cli)
- [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensions)
- [Azure/azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs)

We also have plans to add the process to these repositories: 

- [Azure/azure-sdk-for-c](https://github.com/Azure/azure-sdk-for-c)
- [Azure/azure-sdk-for-cpp](https://github.com/Azure/azure-sdk-for-cpp)
- [Azure/azure-sdk-for-android](https://github.com/Azure/azure-sdk-for-android)
- [Azure/azure-sdk-for-ios](https://github.com/Azure/azure-sdk-for-ios)

> 
> 
> 
> ## A label driven process 

Reviewer: ```suggestion
## A label-driven process 
```

| `needs-triage`      | To indicate that the issue needs to be reviewed by triage team                                               |
| `customer-reported` | To distinguish community created issues from those created by the SDK team for development tracking purposes |
| `question`          | To indicate issue type. (The assumption is that the issue is a question.)                                    |

The triage team will go through the issues and assess who might be the best owner to assign to an issue and whether an issue needs to be routed to another team to be resolved. More labels will be added as necessary or removed if the issue turns out to be about a bug (for example) rather than a question:   

| Tag                      | Purpose                                               |
| ------------------------ | ----------------------------------------------------- |
| `<Service Name>`         | To indicate what service the issue is referring to    |
| `Service Attention`      | Added if the issue needs to be routed to another team |
| `bug`, `feature-request` | These are the other issue types                       |

> For issues that need to be routed to teams best able to assist, i.e. those tagged `Service Attention`, there will be a comment that says the following: 
> 
>     Thanks for the feedback! We are routing this to the appropriate team for follow-up. cc @GitHubHandle
> 
> This will bring attention to the mentioned GitHub handle(s) that they need to help resolve the issue. As the engineers gain more insight into the issue during their investigations, they'll also adjust the labels accordingly. 

Reviewer: ```suggestion
For issues that require help from another team, like those tagged `Service Attention` or `CXP Attention`, there will be a comment that indicates it has been routed for further assistance.   The associated team will be notified and will engage on the issue to investigate and help.  As more insight is gained during investigation, labels will be adjusted.
```

Reviewer: > but not all repo owners leave a comment about routing issue to appropriate team; some just add labels, even for issues that are meant for service teams.

My understanding is that puts us out of SLA.   We were asked during triage to ensure that we comment explicitly unless there's an automated comment.    If we're not seeing that happening during triage - isn't that a process problem that we'll need to figure out?

Reviewer: That said, I've no concerns about leaving that out entirely.    Just mentioning that someone will triage, remove the "needs-triage" tag and apply other tags to get it noticed by those who need to see it sets expectations just fine, IMHO.

> Note that for issues that need routing, sometimes it may take a while to find the right contacts, and so response will be slower than usual. When this happens, we encourage you to leave us a comment. 

Reviewer: The service and CXP team have an SLA of 72 hours for the first "quality" reponse;  we should see someone engage within that window.   If there's no interaction in about a week, that's when it may be a good idea to ping the issue so we can try and get it noticed.

> ### 2. Response and follow ups

Reviewer: ```suggestion
### 2. Response and follow-ups
```

> Depending on the nature of the issue, several rounds of back-and-forth may be needed to resolve the problem after the initial response, which we do our best to provide *within 3 business days* of issue creation. You may also be asked for more information — for instance error logs or code to replicate the problem — to help with the investigation. To facilitate this back-and-forth process, we use two labels, `needs-author-feedback` and `needs-team-attention`. 

Reviewer: My understanding is that the "3 business days" SLA is the measurement of time to first quality response.    Given that issues will require different levels of effort and often require additional information gathered by customers,  the mention of "we'll try to resolve things within 3 days" seems unwise.

@AlexGhiondea - I'd appreciate your thoughts here.

Author: Oh maybe the phrasing isn't clear here -- the initial response is what is provided within 3 business days, not the actual solution.

When asking you for more input, we also add the `needs-author-feedback` label to the issue. When tagged as such, our bot will remind either you or us to look at the issue depending on which side has responded: 
- when you respond, it adds `needs-team-attention` to get our attention (and removes `needs-author-feedback`)
- when you don’t respond *7 days after* `needs-author-feedback` is added, it'll add `no-recent-activity` and send you a friendly reminder

>     - if there’s no response from you after *another 14 days* (so total 21 days), it'll close the issue

Reviewer: ```suggestion
- if there’s no response from you after *another 14 days* (total 21 days), it'll close the issue
```

- if you respond *within 7 days* of issue closure, it'll reopen the issue 

> ## Issue types and when they're resolved

Reviewer: We should include some details of the `issue-addressed` tag and workflow here.    _([wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/6/Issue-Management))_

## Issues not under consideration

Sometimes there are requests or suggestions that we don't feel align with the direction of the product or would positively impact the majority of customers. When that is the case, we close the issue with an explanation. 

There are also cases where an issue is about something that we will not and/or cannot fix — for example, libraries that have been deprecated or endpoints that have been removed. This is especially true for older issues; we've seen 2+ year-old issues about older versions of a product that we no longer have sufficient knowledge on. When that is the case, we close the issue with a brief explanation. 

Finally, it's also possible that the problem brought up in an issue is not considered in the short term. In this case, the issue is put in the backlog. This means other work items are prioritized first, and we're uncertain about when the issue will be resolved. 

## Reopening issues

> Our current support process has room for improvement. Sometimes we or the bot might make the wrong judgment and close an issue prematurely. If that's the case, you're welcome to reopen the issue and ping us to help by leaving a comment. 

Reviewer: Customers don't have rights to reopen issues directly.    The `/unresolve` command included with `issue-addressed` would be the recommendation here;  that will reopen the issue and reset the tags so that it gets attention.
